layer_factory/assistant_layer/mem_bound_warning.cpp

```diff
@@ -0,0 +1,210 @@
+            if (images[i] != VK_NULL_HANDLE) continue;  // Already retrieved image
+            images[i] = pSwapchainImages[i];
+        }
+    }
+
+    return VK_SUCCESS;
+}
+
+// Get memory associated with the handle being destroyed then find handle set using this memory and erase it
+template <typename T>
+void MemBoundWarning::RemoveBindedMemory(T handle) {
+    assert(handle);
+    uint64_t handle_64 = HandleToUint64(handle);
+    auto map_pair = handle_memory_map.find(handle_64);
+    if (map_pair == handle_memory_map.end()) return;  // this will happend if handle was never Binded to memory...prevents segfault
+    auto memory_set = map_pair->second;
+
+    // remove all memory in the memory set. If it is not sparse memory, there will only be one item in set
+    for (VkDeviceMemory memory : memory_set) {
+        auto map_pair = memory_handle_map.find(memory);
+        if (map_pair == memory_handle_map.end())
+            continue;  // this will happend if handle was never Binded to memory...prevents segfault
+        auto &handle_set = map_pair->second;
+        handle_set.erase(handle_64);
+    }
+}
+
+void MemBoundWarning::PostCallDestroySwapchainKHR(VkDevice device, VkSwapchainKHR swapchain,
+                                                  const VkAllocationCallbacks *pAllocator) {
+    assert(swapchain);
+
+    auto map_pair = swapchain_image_map.find(swapchain);
+    if (map_pair == swapchain_image_map.end())
+        return;  // this will happend if handle was never Binded to memory...prevents segfault
+    auto images = map_pair->second;
+
+    if (images.size() > 0) {
+        for (auto swapchain_image : images) {
+            RemoveBindedMemory<VkImage>(swapchain_image);
+        }
+    }
+
+    swapchain_image_map.erase(swapchain);
+}
+
+void MemBoundWarning::PreCallDestroyBuffer(VkDevice device, VkBuffer buffer, const VkAllocationCallbacks *pAllocator) {
+    RemoveBindedMemory<VkBuffer>(buffer);
+}
+
+void MemBoundWarning::PreCallDestroyImage(VkDevice device, VkImage image, const VkAllocationCallbacks *pAllocator) {
+    RemoveBindedMemory<VkImage>(image);
+}
+
+void MemBoundWarning::PreCallFreeMemory(VkDevice device, VkDeviceMemory memory, const VkAllocationCallbacks *pAllocator) {
+    assert(memory);
+    auto map_pair = memory_handle_map.find(memory);
+    if (map_pair == memory_handle_map.end()) return;  // this will happend if handle was never Binded to memory...prevents segfault
+    auto handle_set = map_pair->second;
+
+    for (const auto &handle : handle_set) {
+        char handle_string[64];
+        char memory_string[64];
+        uint64_t handle_64 = HandleToUint64(handle);
+        uint64_t memory_64 = HandleToUint64(memory);
+        sprintf(handle_string, "0x%" PRIxLEAST64, handle_64);
+        sprintf(memory_string, "0x%" PRIxLEAST64, memory_64);
+
+        std::stringstream message;
+        message << "Vk Object " << handle_string << " still has a reference to Memory Object " << memory_string;
```

Comment:
You should be able to avoid the sprintf and operate on the uint64_t values directly with something like:
`message << "Vk Object 0x" << std::hex << handle_64 << " still has a reference to Memory Object 0x" << memory_64;`

Reply:
As per our discussion, we decided this should be left the way it is to stay consistent with validation_layers and for a few other consistency reasons.